BackEnd/src/business/stocks.ts

```diff
@@ -0,0 +1,156 @@
+  let currentTimeStamp = new Date(Date.now());
+
+  // insert new value in timeline for both teams
+  await createDBNewStockValue(winningTeamId, seasonId, updatedStockValues.newWinnerDollarValue, currentTimeStamp);
+  await createDBNewStockValue(losingTeamId, seasonId, updatedStockValues.newLoserDollarValue, currentTimeStamp);
+}
+
+export async function buyStock(user: AuthUser, seasonId: number, teamId: number, amountToBuy: number) {
+  let team: TeamModel = await GetDbTeamsByTeamId(teamId, seasonId);
+  if(!team) {
+    throw new InvalidRequestError(`Could not find team with id ${teamId} and season ${seasonId}`);
+  }
+
+  let userWallet = await getUserWallet(user);
+  let priceOfStock = (await getDbLatestStockValue(seasonId, teamId)).dollarValue;
+  let price = priceOfStock * amountToBuy;
+  if(price < userWallet) {
+    throw new InvalidRequestError('User cant buy that stock');
+  }
+
+  let didWithdrawSucceed = await withdrawDbFunds(user, price);
+  let didBuySucceed =  await buyDbStockForUser(user, seasonId, teamId, amountToBuy);
+  // TODO some success messaging
+  return;
+}
+
+async function getUserWallet(user: AuthUser): Promise<number> {
+  return 0; // TODO
+}
+
+function calculateNewDollarValue(winnerDollarValue: number, loserDollarValue: number, kFactor: number = 250) {
+
+  // Extract current ratings from the team objects
+  const ratingWinner = winnerDollarValue;
+  const ratingLoser = loserDollarValue;
+
+  // Calculate the expected scores
+  const expectedWinner = 1 / (1 + Math.pow(10, (ratingLoser - ratingWinner) / 400));
+  const expectedLoser = 1 / (1 + Math.pow(10, (ratingWinner - ratingLoser) / 400));
+
+  // Update ratings based on the fixed result (winner gets 1, loser gets 0)
+  const newRatingWinner = ratingWinner + kFactor * (1 - expectedWinner);
+  const newRatingLoser = ratingLoser + kFactor * (0 - expectedLoser);
+
+  return {
+    newWinnerDollarValue: Math.round(newRatingWinner),
+    newLoserDollarValue: Math.round(newRatingLoser)
+  };
+}
+
+async function createBaselineForTeam(seasonId: number, teamId: number): Promise<number> {
+  let team = await GetDbTeamRosterByTeamId(teamId, seasonId);
+  let total_elo = 0;
+
+  for (let player of team) {
+    let riotPlayer = await GetRiotPlayerByPuuid(player.playerPuuid);
+    let rank = await  GetRiotLeagueBySummonerId(riotPlayer.id);
+    let elo = mapLeagueRankToElo(rank);
+    total_elo += elo;
+  }
+
+  return total_elo / team.length;
+}
+
+
+function mapLeagueRankToElo(rankInformation: RiotLeagueEntryDto): number {
+  let rankMap: Record<string, number> =  {
+    'I': 375,
+    'II': 250,
+    'III': 125,
+    'IV': 0,
+  };
+
+  let tierMap: Record<string, number> = {
+    'IRON': 0,
+    'BRONZE': 500,
+    'SILVER': 1000,
+    'GOLD': 1500,
+    'PLATINUM': 2000,
+    'EMERALD': 2500,
+    'DIAMOND': 3000,
+    'MASTER': 3500,
+    'GRANDMASTER': 4000,
+    'CHALLENGER': 4500
+  };
+
+  // Default to PLATINUM elo if the player has not played rank yet
```

Comment:
thankfully this should never happen because a player has game reqs

Reply:
Not necessarily, take this split of risen. You didnt need any games to start the season. So if a player is in placements this would return null.

Reply:
Oh, in that case I think rather than use Plat as a default we should remove them from the calc